assets/js/components/Dashboard/NewPaMessage/SelectStationsPage.tsx

```diff
@@ -0,0 +1,278 @@
+              value={zones}
+              onChange={setZones}
+            />
+            <StationGroupCheckbox
+              title="Ashmont Branch"
+              label="JFK-Ashmont"
+              places={places}
+              routes={["Red"]}
+              stations={RED_ASHMONT_BRANCH}
+              value={zones}
+              onChange={setZones}
+            />
+            <StationGroupCheckbox
+              title="Red Line Trunk"
+              label="Alewife-JFK"
+              places={places}
+              routes={["Red"]}
+              stations={RED_TRUNK}
+              value={zones}
+              onChange={setZones}
+            />
+            <hr />
+          </div>
+          <div className="route-groups">
+            <div className="h5">Orange line</div>
+            <StationGroupCheckbox
+              title="North"
+              label="Oak Grove-North Station"
+              places={places}
+              routes={["Orange"]}
+              stations={ORANGE_NORTH}
+              value={zones}
+              onChange={setZones}
+            />
+            <StationGroupCheckbox
+              title="South"
+              label="Haymarket-Forest Hills"
+              places={places}
+              routes={["Orange"]}
+              stations={ORANGE_SOUTH}
+              value={zones}
+              onChange={setZones}
+            />
+          </div>
+        </div>
+        <div className="route-col route-col--green col">
+          <Form.Check
+            className="title"
+            label="Green line"
+            type="checkbox"
+            id="green-line"
+            onChange={(evt) => {
+              if (evt.target.checked) {
+                setZones(fp.union(greenLineZones, zones));
+              } else {
+                setZones(fp.without(greenLineZones, zones));
+              }
+            }}
+            checked={greenLineZones.every((zone) => zones.includes(zone))}
+          />
+
+          <div className="branches-col">
+            {["B", "C", "D", "E"].map((branch) => {
+              // It's possible to check a station on one branch even though not all zones are checked.
+              // If you want to instead select all branches when selecting a station,
+              // pass GREEN_LINE_ROUTES to routes and pass route to orderingRoute
+              const route = `Green-${branch}`;
+              return (
+                <RouteColumn
+                  key={branch}
+                  label={`${branch} branch`}
+                  routes={[route]}
+                  places={placesByRoute[route]}
+                  value={zones}
+                  onChange={setZones}
+                />
+              );
+            })}
+          </div>
+        </div>
+
+        {["Red", "Orange", "Blue", "Mattapan", "Silver"].map((route) => (
+          <div
+            key={route}
+            className={`route-col route-col--${route.toLowerCase()} col`}
```

Comment:
**quibble:** i'm not a fan of dynamically building CSS classes because i think it makes it very difficult to determine where a particular CSS class is used in our markup.

**suggestion:** having a mapping of `route` to values that look like `route-col--red`, `route-col--blue`, etc. would make it so you could search for `route-col--red` and find where we use that value